### DISCUSSION TITLE
Migrations with Beanie... out-of-the-box or only with replica sets?

### DISCUSSION THREAD

**Discussion by palminha** [BeanieODM/beanie#338] [Question]
I've read that beanie supports out-of-the-box migrations of data and schema...

but in the official documentation ( https://roman-right.github.io/beanie/tutorial/migrations/ ) is stated that _Migrations use transactions inside. It works only with MongoDB replica sets_

is the documentation up-to-date? does the migration really only work with a MongoDB replica set?

**Comment by roman-right** [comment]
Hey!
It works only for replica sets to be able to roll the changes back if something went wrong during the transaction.

I didn't check the new version of mongo, probably it supports transactions without RS - I'll let you know here if this is not a requirement and will update the migrations code respectively.
